.github/actions/prepare-instance/action.yml

```diff
@@ -68,3 +68,9 @@ runs:
           --domain="$INSTANCE_NAME" \
           --skip-restrict \
           --skip-webhooks-update
+
+    - name: Add client origins
+      shell: bash
+      if: ${{!contains(inputs.BASE_URL, inputs.POOL_INSTANCE) && !steps.instance_check.outputs.INSTANCE_KEY }} # Add base url as a trusted origins if we're using deployed dashboard in repo (saleor.rocks)
+      run: |
+        npx saleor env origins ${{ inputs.POOL_NAME }} --origins=${{ inputs.BASE_URL }}
```

Comment:
Do not use template interpolation inside shell blocks as they may cause malformed shell scripts which can potentially be abused.

There were also missing quotes which could cause shell expansion if there were whitespaces inside the values, e.g.,:

```
~/workdir $ origins="https://foo.bar blah"

# Bad because it the following (it split the values and we don't want that): saleor env origins --origins 'https://foo.bar' 'blah' 'myenv'
~/workdir $ saleor env origins --origins $origins myenv
saleor env origins [key|environment]

Manage environment's trusted client origins

Positionals:
  key, environment, environment  key of the environment  [string]

Options:
      --json             Output the data as JSON  [boolean] [default: false]
[...]

# Good because it doesn't split: saleor env origins --origins 'https://foo.bar blah' 'myenv'
~/workdir $ saleor env origins --origins "$origins" myenv
✔ Organization · mikails-organization-1
```

There are more details about word splitting here: https://www.shellcheck.net/wiki/SC2086


```suggestion
      if: ${{!contains(inputs.BASE_URL, inputs.POOL_INSTANCE) && !steps.instance_check.outputs.INSTANCE_KEY }} # Add base url as a trusted origins if we're using deployed dashboard in repo (saleor.rocks)
      env:
        POOL_NAME: ${{ inputs.POOL_NAME }}
        BASE_URL: ${{ inputs.BASE_URL }}
      run: |
        npx saleor env origins "$POOL_NAME" --origins="$BASE_URL"
```

Reply:
AFAIK you reviewed these changes added by @karola312 in https://github.com/saleor/saleor-dashboard/pull/4904 , but I will commit your suggestion as requested.